### DISCUSSION TITLE
Which dart/flutter versions are supported (until when)?

### DISCUSSION THREAD

**Discussion by karfau** [codemagic-ci-cd/codemagic-docs#1759] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

This seems to be the best spot for my question, so I'll put it here.

We are aware that we are lagging behind in updating our app regarding dart (v2) and flutter (v3.3) versions as well as some dependency versions.

To know how much time pressure we have until we need to get this fixed, we would like to know how long Dart v2 and older flutter versions will be supported by codemagic.

### Expected results

getting an answer to my question above

### Actual results

not knowing the answer

**Comment by himesh-cm** [comment]
Hi @karfau,

Please feel free to use any version of Flutter that suits your needs. We will update the pre-installed version on the host as and when it is released.

To manage flutter versions on codemagic, specify it in the `yaml` file, select a version from the dropdown in the `workflow editor` settings, or use FVM. 
For more information, please refer to our documentation [here](https://docs.codemagic.io/yaml-quick-start/building-a-flutter-app/#setting-the-flutter-version).

**Reply by karfau** [reply]
I'm just aware that we are specifying the dart sdk version (range) in the `pubspec.yaml` and the flutter version in the `codemagic.yml` and there is for sure some matrix of flutter and dart versions working together (which might be what is pointed out here: https://docs.flutter.dev/release/archive?tab=linux, but not sure).

Do you ignore the version that is specified in `pubspec.yml` and always pick the one that comes with the flutter version one picks in `codemagic.yml`?

And another related question: we received a mail from apple this morning after we created a new app version that we seem to have used `altool` :+1: 

> We’re reaching out because you recently used the altool command-line utility to notarize your macOS software with Apple. As announced last year at [WWDC22](https://developer.apple.com/videos/play/wwdc2022/10109), if you’re still using altool with the Apple notary service, you should transition to the notarytool command-line utility as soon as possible. Notarizing software with altool was deprecated in Xcode 13, and the Apple notary service will no longer accept uploads from altool as of November 1, 2023. Existing notarized software will continue to function properly. 
>
>For information on notarizing your apps, read [TechNote TN3147: Migrating to the latest notarization tool](https://developer.apple.com/documentation/technotes/tn3147-migrating-to-the-latest-notarization-tool?changes=_3_3). 
>
>If you have any questions, [contact us](https://developer.apple.com/contact/).

I still find `altool` in some of places in the docs of codemagic/flutter, but not in our codebase and no mentions of `notarytool` anywhere. Would we need to upgrade to a newer flutter version (do you happen to know which one?) to switch to the new suggested `notarytool`?

I will have a look at what `fvm` can do since this might be a good way to handle it for both dev machines and codemagic in the same way.

**Reply by himesh-cm** [reply]
Hi @karfau, The version of Dart will be determined based on the Flutter version specified in the `yaml` file. 

The warning message is specifically for macOS applications. Are you also attempting to build on a Mac?

**Reply by karfau** [reply]
As far as I'm aware we are only building for iOS not for macOS.
But of course we are doing that on a macos in codemagic...
Did that answer your question?

**Reply by himesh-cm** [reply]
You can ignore the `notarytool` warning in that case.

**Reply by karfau** [reply]
Let's hope that works...

**Comment by himesh-cm** [comment, marked as answer]
While I’m closing our thread for now, if you have any other questions or suggestions, feel free to reply back. Have a nice day!
